axelor-sale/src/main/resources/views/CartLine.xml

```diff
@@ -5,14 +5,18 @@
 
   <grid name="cart-line-grid" title="Cart lines" model="com.axelor.apps.sale.db.CartLine"
     editable="true">
-    <field name="product"/>
+    <field name="product" form-view="product-form" grid-view="product-grid"/>
+    <field name="product.code"/>
     <field name="qty"/>
+    <field name="product.salePrice"/>
   </grid>
 
   <form name="cart-line-form" title="Cart line" model="com.axelor.apps.sale.db.CartLine">
     <panel name="mainPanel">
       <field name="product" form-view="product-form" grid-view="product-grid"/>
+      <field name="product.code"/>
       <field name="qty"/>
+      <field name="product.salePrice"/>
```

Comment:
Please add a scale for product.salePrice. The value must be fetched from appBase : nbDecimalDigitForUnitPrice.

Also on the form and grid views, place the product.salePrice after product.code.